crates/vault-registry/src/lib.rs

```diff
@@ -272,49 +237,48 @@ decl_module! {
         /// # Errors
         /// * `VaultNotFound` - if no vault exists for the origin account
         /// * `InsufficientCollateralAvailable` - if the vault does not own enough collateral
-        #[weight = <T as Config>::WeightInfo::withdraw_collateral()]
+        #[pallet::weight(<T as Config>::WeightInfo::withdraw_collateral())]
         #[transactional]
-        fn withdraw_collateral(origin, #[compact] amount: Backing<T>) -> DispatchResult {
+        pub fn withdraw_collateral(
+            origin: OriginFor<T>,
+            #[pallet::compact] amount: Backing<T>,
+        ) -> DispatchResultWithPostInfo {
```

Comment:
Is `DispatchResultWithPostInfo` now required for all calls?

Reply:
According to the [upgrade guidelines](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#upgrade-guidelines), yes